drivers/mtd_spi_nor/mtd_spi_nor.c

```diff
@@ -378,6 +400,11 @@ static int mtd_spi_nor_power(mtd_dev_t *mtd, enum mtd_power_state power)
                 return -EIO;
             }
 #endif
+            /* enable 32 bit address mode */
+            if (dev->params->addr_width == 4) {
```

Comment:
Would an additional `assert(dev->params->addr_width == 3 || dev->params->addr_width == 4)` help catch configuration issues?

Reply:
Sure, can add.
Initially I wanted to allow for setting `addr_width = 0` in the config for auto detection (size > 16 MiB -> 4 byte addressing, otherwise 3 byte) but the `params` struct is read only.